vortex-array/src/array/bool/mod.rs

```diff
@@ -127,19 +124,15 @@ impl BoolArray {
     pub fn try_new(buffer: BooleanBuffer, validity: Validity) -> VortexResult<Self> {
         let buffer_len = buffer.len();
         let buffer_offset = buffer.offset();
-        let first_byte_bit_offset = (buffer_offset % 8) as u8;
-        let buffer_byte_offset = buffer_offset - (first_byte_bit_offset as usize);
 
-        let inner = buffer
-            .into_inner()
-            .bit_slice(buffer_byte_offset, buffer_len);
+        let inner = buffer.into_inner().bit_slice(buffer_offset, buffer_len);
```

Comment:
arrow changed the behavior of `bit_slice` in cases where the slicing is aligned, see my other comment.

Reply:
the idea here was that we slice to the byte offset since those are "free" and if we slice exactly to the offset we will end up making a copy potentially

Reply:
yeah put that in the other comment, I'll put that back in on Monday or if I get bored over the weekend